app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -399,7 +424,67 @@ public class MainActivity : AppCompatActivity(), ViewController,
             (findViewById(R.id.route_preview) as RoutePreviewView).destination =
                     SimpleFeature.fromFeature(destination);
             (findViewById(R.id.route_preview) as RoutePreviewView).route = route;
+            displayRoute(route)
+
         })
+        updateRoutePreview(destination)
+    }
+
+    fun displayRoute(route: Route?) {
+        try {
+            mapController!!.getMap().layers().remove(path)
+            mapController!!.getMap().layers().remove(markers)
+            path = PathLayer(mapController!!.getMap(), Color.DKGRAY, 8f)
+            markers = ItemizedLayer(mapController!!.getMap(), ArrayList<MarkerItem>(), AndroidGraphics.makeMarker(this.getResources().getDrawable(R.drawable.ic_pin), MarkerItem.HotspotPlace.BOTTOM_CENTER), null)
```

Comment:
I haven't setup checkstyle yet (not sure if there is a Kotlin version for this?) but lines over 100 characters should be wrapped.